WordPress/Classes/ViewRelated/Post/PageMenuViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+
+final class PageMenuViewModel: AbstractPostMenuViewModel {
+
+    private let page: Page
+    private let isJetpackFeaturesEnabled: Bool
+    private let isBlazeFlagEnabled: Bool
+
+    var buttonSections: [AbstractPostButtonSection] {
+        [
+            createPrimarySection(),
+            createSecondarySection(),
+            createBlazeSection(),
+            createTrashSection()
+        ]
+    }
+
+    init(
+        page: Page,
+        isJetpackFeaturesEnabled: Bool = JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled(),
+        isBlazeFlagEnabled: Bool = BlazeHelper.isBlazeFlagEnabled()
+    ) {
+        self.page = page
+        self.isJetpackFeaturesEnabled = isJetpackFeaturesEnabled
+        self.isBlazeFlagEnabled = isBlazeFlagEnabled
+    }
+
+    private func createPrimarySection() -> AbstractPostButtonSection {
+        var buttons = [AbstractPostButton]()
+
+        if !page.isFailed {
+            buttons.append(.view)
+        }
+
+        return AbstractPostButtonSection(buttons: buttons)
+    }
+
+    private func createSecondarySection() -> AbstractPostButtonSection {
+        var buttons = [AbstractPostButton]()
+
+        if page.status != .draft {
+            buttons.append(.moveToDraft)
+        }
+
+        if page.status == .publish || page.status == .draft {
+            buttons.append(.duplicate)
+        }
+
+        if page.status != .trash && page.isFailed {
+            buttons.append(.retry)
+        }
+
+        if !page.isFailed, page.status != .publish && page.status != .trash {
+            buttons.append(.publish)
+        }
+
+        return AbstractPostButtonSection(buttons: buttons)
+    }
+
+    private func createBlazeSection() -> AbstractPostButtonSection {
+        var buttons = [AbstractPostButton]()
+
+        if isBlazeFlagEnabled && page.canBlaze {
+            BlazeEventsTracker.trackEntryPointDisplayed(for: .pagesList)
```

Comment:
In the production version, this tracking event is called when the action sheet is displayed. In the new version, it's' called when the menu is created. I would suggest either moving it to [`UIControlEventMenuActionTriggered `](https://developer.apple.com/documentation/uikit/uicontrolevents/uicontroleventmenuactiontriggered) or trying to make the menu lazy. The latter would probably be a better option since it would also reduce the amount of work on the main thread when the cells are displayed.